src/packages/src/xroad/common/proxy/usr/share/xroad/scripts/backup_xroad_proxy_configuration.sh

```diff
@@ -20,9 +20,21 @@ OPTIONS:
     -s ID of the security server.
     -f Absolute path of the resulting tar archive.
     -S Skip database backup
+    -E encrypt backup
 EOF
 }
 
+get_proxy_prop() {
+  # local.ini overrides keys
+  local value=$(crudini --get /etc/xroad/conf.d/local.ini "$2" "$3" 2>/dev/null || echo EMPTYKEY)
+  if [ -n "$value" ]; then
+    if [ $value = EMPTYKEY ]; then
+      local value=$(crudini --get /etc/xroad/conf.d/"$1" "$2" "$3" 2>/dev/null || echo -n "$4")
+    fi
+  fi
+  echo $value
+}
```

Comment:
Alternatively, avoids treating EMPTYKEY a special value
```
get_proxy_prop() {
  local value
  if ! value=$(crudini --get /etc/xroad/conf.d/local.ini "$2" "$3" 2>/dev/null); then
    value=$(crudini --get /etc/xroad/conf.d/"$1" "$2" "$3" 2>/dev/null || echo "$4")
  fi
  echo "$value"
}
```

Reply:
Yes this is definitely better, my bash skills are really lacking.